### PR TITLE
added ioc container - feature_16

### PR DESCRIPTION
**Pull request by pollend**
I reconfigured the data layer to work using an ioc container. I think this should be more maintainable in the long run. I haven't re-implemented the container to work across the forms themselves yet. At the moment i'm just hacking in the dependencies to the get the client to work. I haven't ported the linux and mac client yet. I would place this in a feature branch and then work on developing out this feature completely. presumably I might have introduced a couples regression given how extensive these changes are. 

How great, your new windows client is perfectly setup for the ioc container compared to the previous client.

It appears everything is now working with the new windows client. I just had to reconfigure the logger a bit and everything appears to be working perfectly.

### CODE REVIEW

**Comment by ikkentim** [comment]
I'm going to have to look into this package a bit. I'm not unfamiliar to IOC contains in other environments. In the meanwhile, can you carefully review your changes? https://github.com/ParkitectNexus/ParkitectNexusClient/pull/18/files I'm sure you'll be able to spot the hiccups I've spotted too.

**Comment by pollend** [comment]
Yep, I have to clean this up a bit. I've been kind of losing tracking of what I've been changing.

**Comment by ikkentim** [comment]
Don't think that last commint was supposed to be part of this PR

`started developing new nexus client for linux …`

**Comment by ikkentim** [comment]
Also, you're using tabs again  :) Please review your code before you commit it :)

**Comment by pollend** [comment]
I don't think there is a way to make the another pull request that pull's in changes from this branch. I break the build so in fairness i think this would fit in with this pull request.

**Comment by ikkentim** [comment]
You could simply create multiple branches. It's good practice to create separate branches for different PRs (eg patch-1, patch-2 or feature/X, feature/Y)

File: src/ParkitectNexus.Client.Windows/Resources/.directory [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[Dolphin]
```
Comment (ikkentim) [inline src/ParkitectNexus.Client.Windows/Resources/.directory]: Whoops?

**Comment by pollend** [comment]
there you go i rewrote the history and moved the linux client to feature 11

File: src/ParkitectNexus.Client.Windows/Resources/.directory (hunk 1 above)
Comment (pollend) [inline src/ParkitectNexus.Client.Windows/Resources/.directory]: Thanks, I create a global ignore.

**Comment by ikkentim** [comment]
I'd merge it except for this, http://puu.sh/mquJD/acac9fa3c7.png

**Comment by ikkentim** [comment]
Nvm, I'll fix it myself

**Comment by ikkentim** [comment]
I checked it out and fixed some compile-time errors: https://github.com/ParkitectNexus/ParkitectNexusClient/commit/6ef538fd44bdaca5ea821e6470983adc1df63e5f

But it still has an issue. The Repository factory keeps creating new instances of repository, and somehow the settings don't save. 

I can't pick my installation path, because every time the WindowsParkitect.InstallationPath is set, it's not stored in the config, and next time it tries to read the installation path it's null again

**Comment by pollend** [comment]
opps, i forgot to bring in that change from the other branch.

**Comment by pollend** [comment]
This should fix your loading issue. problem is that it was saving the repository itself so the save wouldn't match the load.

**Comment by ikkentim** [comment]
Oh wow how did I not see that

**Comment by ikkentim** [comment]
The other problem still stands though :-( If you place a breakpoint in the Repository constructor, you'll see that it is initialized 15 times while it should only be initialized twice (ClientSettings/GameSettings). Somehow the instances aren't kept alive

**Comment by ikkentim** [comment]
Oh I see, the type needs to be marked singleton in the registry. I'll do that.

**Comment by pollend** [comment]
maybe if you use a scanner and scan models that include an IModel interface and mark those as singleton. 

  Scan(x =>
            {
                x.IncludeNamespaceContainingType<IModel>();
                x.WithDefaultConventions();
                x.SingleImplementationsOfInterface();
            });

**Comment by ikkentim** [comment]
Yea, I'll just merge it and create an issue.

**Pull request merged**